treewide: Fix links in the docs and cleanup many boards (#22257)
RIOT-OS/RIOT: boards/msbiot/doc.md

Reviewer: ```suggestion
```

Reviewer: Same here, the section can be removed.

#### 3 user controllable LEDs

| NAME      | LED_RED       | LED_YELLOW   | LED_GREEN    |
| --------- | ------------- | ------------ | ------------ |
| Color     | red           | yellow       | green        |
| Pin       | PB8           | PB14         | PB15         |
| Pin Alias | LED0_PIN      | LED1_PIN     | LED2_PIN     |
| Macros    | LED0_ON, ...  | LED1_ON, ... | LED2_ON, ... |

The LEDs can be controlled inside your RIOT application by using the LED
macros LED0_ON, LED0_OFF, and LED0_TOGGLE for the red LED,
LED1_ON, LED1_OFF, and LED1_TOGGLE for the yellow LED, and

> LED2_ON, LED2_OFF, and LED2_TOGGLE for the green LED.

Reviewer: ```suggestion
LED2_ON, LED2_OFF, and LED2_TOGGLE for the green LED.
See our [API documentation](https://api.riot-os.org/group__drivers__led.html) for
more information about the standard LED macros.
```

| SDA                   | PB7                                                                                               |
| IRQ Line              | PB11                                                                                              |

### Other components

For the following components of the MSB-IoT, there is currently no support in
RIOT:

* [TCA6416 16-Bit I/O Expander](http://www.ti.com/product/tca6416)
* Micro-SD-Card Interface

## Compiling, Flashing and Debugging

This section only covers the automated functionality provided by RIOT's
makefile system. If you don't want to rely on RIOT's make targets for flashing
or debugging (for instance to use a different flashing tool), please check the
documentation of your used tools.

### Flashing

RIOT's provided functionality for debugging and flashing the MSB-IoT is based

> on the [Open On-Chip Debugger](http://openocd.org/) tool. A wiki page with

Reviewer: ```suggestion
on the [Open On-Chip Debugger](http://openocd.org/) tool. A guide with
```

> RIOT's provided functionality for debugging and flashing the MSB-IoT is based
> on the [Open On-Chip Debugger](http://openocd.org/) tool. A wiki page with
> installation instructions and some other information can be found

Reviewer: ```suggestion
```

Moved.

> [here](https://doc.riot-os.org/misc/openocd/).

Reviewer: ```suggestion
```

> [here](https://github.com/RIOT-OS/RIOT/wiki/OpenOCD).

Reviewer: This can also be replaced by a reference to https://doc.riot-os.org/board_specific/stm32/
Perhaps also relevant for other boards?

Also: TIL that we had a wiki site about OpenOCD. If you want to, you can move some of the relevant information from the Wiki site to the Guide site too and replace the Wiki site with a link to said guides.

Perhaps even make a dedicated OpenOCD guide? 👀 
_**Scope creeeeeeeep!**_

Author: Done Sir

Reviewer: Thank you :)

Author: <img width="500" height="375" alt="image" src="https://github.com/user-attachments/assets/e67264da-3152-462a-8ee8-65f84fbfa2cb" />

/home/fabian/myriot/RIOT/examples/basic/hello-world/bin/msbiot/hello-world.elf
```

> 

Reviewer: ```suggestion
```

I would get rid of that entirely, also the traces with paths are not relevant.

Author: The commit suggestion was broken, so I hope I removed what you want to remove?

Reviewer: Yes, the whole `Compiling` subsection.

Done flashing

> ```

Reviewer: ````suggestion
### Flashing

RIOT's provided functionality for debugging and flashing the MSB-IoT is based
on the [Open On-Chip Debugger](http://openocd.org/) tool. A wiki page with
installation instructions and some other information can be found
[here](https://doc.riot-os.org/misc/openocd/).

You can flash the board by calling the following command:

```shell
BOARD=msbiot make flash
```
````

(gdb)

> ```

Reviewer: ```suggestion
```

Not board specific => gone.

Author: The commit suggestion was broken, so I hope I removed what you want to remove?

Reviewer: Yes, the whole `Debugging` subsection.
